### PR TITLE
Disallow setting Switch.System.ServiceModel.DisableUsingServicePointManagerSecurityProtocols to true

### PR DESCRIPTION
**Pull request by dotpaul**
Generalizing DoNotDisableSchUseStrongCrypto analyzer.

### CODE REVIEW

**Comment by dotpaul** [comment]
FYI @LingxiaChen @DSIE63 @timhermann @matt-desai

File: src/Microsoft.NetCore.Analyzers/UnitTests/Security/DoNotDisableUsingServicePointManagerSecurityProtocols.cs [hunk 1]
```diff
@@ -0,0 +1,179 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.Diagnostics;
+using Test.Utilities;
+using Xunit;
+
+namespace Microsoft.NetCore.Analyzers.Security.UnitTests
+{
+    public class DoNotDisableUsingServicePointManagerSecurityProtocolsTests : DiagnosticAnalyzerTestBase
```
Comment (LLLXXXCCC) [inline src/Microsoft.NetCore.Analyzers/UnitTests/Security/DoNotDisableUsingServicePointManagerSecurityProtocols.cs]: You named the file as "DoNotDisableUsingServicePointManagerSecurityProtocols.cs", add "Tests"?

File: src/Microsoft.NetCore.Analyzers/Core/Security/DoNotSetSwitch.cs [hunk 2]
```diff
@@ -27,18 +33,38 @@ public sealed class DoNotDisableSchUseStrongCrypto : DiagnosticAnalyzer
...
+            s_Title,
+            s_Message,
+            DiagnosticCategory.Security,
+            DiagnosticHelpers.DefaultDiagnosticSeverity,
+            isEnabledByDefault: DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX,
+            description: s_Description,
+            helpLinkUri: null,
+            customTags: WellKnownDiagnosticTags.Telemetry);
+        internal static DiagnosticDescriptor DoNotDisableSpmSecurityProtocolsRule = SecurityHelpers.CreateDiagnosticDescriptor(
+            "CA5378",
```
Comment (mavasani) [inline src/Microsoft.NetCore.Analyzers/Core/Security/DoNotSetSwitch.cs]: Nit: Can we be consistent in use of fields vs inlining the argument values between the two descriptors?

**Comment by dotpaul** [comment]
Thanks @mavasani !

**Pull request merged**